### PR TITLE
Add to collections should only show editable collections

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: spec/controllers/curation_concerns/collections_controller_spec.rb [hunk 1]
```diff
@@ -245,11 +245,20 @@
...
+      create(:collection, :public)
+      sign_in user
+    end
 
-    it 'does not show flash' do
+    it 'is successful' do
       get :edit, params: { id: collection }
       expect(flash[:notice]).to be_nil
+      # a list of collections I can add items to:
+      expect(assigns[:user_collections].map(&:id)).to match_array [collection.id,
```
Comment (cam156) [inline spec/controllers/curation_concerns/collections_controller_spec.rb]: @jcoyne wasn't collection not supposed to show up?

Comment (jcoyne) [inline spec/controllers/curation_concerns/collections_controller_spec.rb]: The collection created on 252 does not show up. The two that match are on line 25 and on line 248.

Comment (jcoyne) [inline spec/controllers/curation_concerns/collections_controller_spec.rb]: The collection created on 252 does not show up. The two that match are on line 25 and on line 248.

**Comment by jcoyne** [comment]
@cam156 added the test in the second commit that moves the CollectionsService from Sufia to CC.

File: app/services/curation_concerns/collections_service.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    def search_results(access)
+      builder = list_search_builder(access)
+      response = context.repository.search(builder)
+      response.documents
+    end
+
+    private
+
+      def list_search_builder(access)
+        list_search_builder_class.new(context).tap do |builder|
```
Comment (atz) [inline app/services/curation_concerns/collections_service.rb]: This section we would hope to replace when the upstream hydra-head blocked by browse_everything is finally unblocked.  The whole method can then just go away and be replaced by a constructor call.

**Comment by mjgiarlo** [comment]
I believe all concerns expressed on this PR have been addressed. Merging!

**Pull request merged**